### PR TITLE
[tests] Add test for path segments when `go.mod` is present

### PR DESCRIPTION
**Pull request by TooTallNate**
Path segment previously were not working when a `go.mod` file is present. This was fixed likely by https://github.com/vercel/vercel/pull/8291, so now this PR is just a test to ensure this case stays working.

### CODE REVIEW

**Comment by southpolesteve** [comment]
@TooTallNate @styfle can we get this merged? What's the current status?

**Comment by bjartek** [comment]
From my understanding this is still just a failing test. I have not seen any activity after first commits.

File: packages/go/test/fixtures/21-square-brackets-with-go-mod/probes.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "probes": [
+    {
+      "path": "/api/[hello].go",
+      "mustContain": "hello:RANDOMNESS_PLACEHOLDER"
+    },
+    { "path": "/api/sub/[hi].go", "mustContain": "hi:RANDOMNESS_PLACEHOLDER" }
```
Comment (styfle) [inline packages/go/test/fixtures/21-square-brackets-with-go-mod/probes.json]: Is this correct for zero config? I wouldn’t expect the file extension in the path name. And the brackets should be removed.

**Review by cb1kenobi — APPROVED** [review]
LGTM

**Review by styfle — APPROVED** [review]

**Pull request merged**